src/utils/management/CustomState.ts

```diff
@@ -0,0 +1,34 @@
+export class CustomState<T> {
+  private error: string;
+  private pending: boolean;
+  private data: T;
+  private readonly preparedData: T;
+
+  constructor(preparedData: T) {
```

Comment:
You can make this implementation a little bit easier.

First of all remove all private keywords - we need this information outside this class instance for UI purposes. So private field is bad idea.

Remove readonly property prepadedData - we need only data.

In constructor write constructor(public data: T) {} - which is shorthand syntax.

Inside methods ok, init, fail - instead of copy object and creating new object via "new keyword" use::
{ ...this, yourOverridedAttributesHere }.

This will reduce boilerplate code.

Also you can create ofcourse helper private function for creating new object.